source/main.cpp

```diff
@@ -247,6 +249,13 @@ void GameLoop(PlayerInfo &player, Conversation &conversation, bool &debugMode)
 			showCursor = shouldShowCursor;
 			SDL_ShowCursor(showCursor);
 		}
+		bool inMap = !gamePanels.IsEmpty() && (dynamic_cast<MapPanel*>(gamePanels.Top().get()) != nullptr);
+		bool inPlayerInfo = !gamePanels.IsEmpty() && (dynamic_cast<PlayerInfoPanel*>(gamePanels.Top().get()) != nullptr);
```

Comment:
From the [style guide](https://endless-sky.github.io/styleguide/styleguide.xml):
![image](https://user-images.githubusercontent.com/17688683/88479627-8a087500-cf1e-11ea-9b23-75918da39c4f.png)
Can these dynamic_casts be avoided?
(I haven't dealt with the panels before, and I haven't looked too deeply into them so this suggestion might be useless, but could you perhaps add a function to Panel.cpp that is implemented by the other panels that returns a string or enum telling us what panel it is? Then instead of using dynamic_cast, we would just call this function from the top panel and set a boolean to true if the top panel is the Map or PlayerInfo.)

Reply:
> Can these dynamic_casts be avoided?
> could you perhaps add a function to Panel.cpp that is implemented by the other panels that returns ...

Yes, I used the solution from @NomadicVolcano from #5088 and added an "AllowFastForward" virtual function to the panels. The function returns true for the MainPanel, MapPanel, PlayerInfoPanel and returns false for all other panels.